### PR TITLE
[CONTINT-3947] Retry failed host sbom scans

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR modifies the SBOM scanner to also retry host scans

### Describe how to test/QA your changes
The unit test should be enough.
Otherwise, we need to recreate a case where the host scan returns an error and make sure it recovers with an exponential backoff. I am not sure how to do this.

### CODE REVIEW

File: pkg/sbom/scanner/scanner_test.go [hunk 1]
```diff
@@ -90,52 +90,66 @@ func TestRetryLogic_Error(t *testing.T) {
...
-	mockCollector := collectors.NewMockCollector()
-	resultCh := make(chan sbom.ScanResult, 1)
-	errorResult := sbom.ScanResult{Error: errors.New("scan error")}
-	expectedResult := sbom.ScanResult{Report: mockReport{id: imageID}}
-	mockCollector.On("Options").Return(sbom.ScanOptions{})
-	mockCollector.On("Scan", mock.Anything, mock.Anything).Return(errorResult).Twice()
-	mockCollector.On("Scan", mock.Anything, mock.Anything).Return(expectedResult).Once()
-	mockCollector.On("Channel").Return(resultCh)
-	shutdown := mockCollector.On("Shutdown")
-	mockCollector.On("Type").Return(collectors.ContainerImageScanType)
```
Comment (AliDatadog) [inline pkg/sbom/scanner/scanner_test.go:104]: Here we were testing only `collectors.ContainerImageScanType`

File: pkg/sbom/scanner/scanner.go [hunk 2]
```diff
@@ -288,17 +288,15 @@ func (s *Scanner) performScan(ctx context.Context, request sbom.ScanRequest, col
...
 		telemetry.SBOMFailures.Inc(request.Collector(), request.Type(), "nil_scan_result")
 		log.Errorf("nil scan result for '%s'", request.ID())
 		return
 	}
 	if scanResult.Error != nil {
 		telemetry.SBOMFailures.Inc(request.Collector(), request.Type(), errorType)
-		if collector.Type() == collectors.ContainerImageScanType {
-			s.scanQueue.AddRateLimited(request)
-		}
+		s.scanQueue.AddRateLimited(request)
```
Comment (AliDatadog) [inline pkg/sbom/scanner/scanner.go:299]: Removed the `if` here that was not called for host sboms.

**Review by davidor — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Comment by AliDatadog** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Comment by AliDatadog** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Comment by AliDatadog** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on

**Pull request merged**